lighthouse-core/config/index.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const defaultConfig = require('./default.json');
+const recordsFromLogs = require('../lib/network-recorder').recordsFromLogs;
+const CriticalRequestChainsGatherer = require('../driver/gatherers/critical-request-chains');
+const Driver = require('../driver');
+const log = require('../lib/log');
+
+function filterPasses(passes, audits) {
```

Comment:
yes, but `passes.map(pass => { /* mutate pass */; return pass; })` keeps the same pass object so the new `gatherers` array is on the old `pass` object in a new `passes` array, so the old `passes` array will also have the new `gatherers` arrays on its elements